xonsh/ptk_shell/shell.py

```diff
@@ -205,7 +204,7 @@ def __init__(self, **kwargs):
             ptk_args.setdefault("clipboard", PyperclipClipboard())
         self.prompter: PromptSession = PromptSession(**ptk_args)
 
-        self.prompt_formatter = PTKPromptFormatter(self.prompter)
+        self.prompt_formatter = PTKPromptFormatter(self)
```

Comment:
What do these changes have to do with the lazy loading?

Reply:
it needs to call a method on shell class